mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndex.java

```diff
@@ -1339,11 +1351,11 @@ public Mono<Void> deleteAll(MailboxSession session, MailboxId mailboxId) {
     }
 
     public void delete(MailboxId mailboxId, MessageRange range) throws IOException {
-        BooleanQuery query = new BooleanQuery();
-        query.add(new TermQuery(new Term(MAILBOX_ID_FIELD, mailboxId.serialize())), BooleanClause.Occur.MUST);
-        query.add(createQuery(range), BooleanClause.Occur.MUST);
+        BooleanQuery.Builder queryBuilder = new BooleanQuery.Builder();
+        queryBuilder.add(new TermQuery(new Term(MAILBOX_ID_FIELD, mailboxId.serialize())), BooleanClause.Occur.MUST);
+        queryBuilder.add(createQuery(range), BooleanClause.Occur.MUST);
 
-        writer.deleteDocuments(query);
+        writer.deleteDocuments(queryBuilder.build());
```

Comment:
Shall we also explicitly delete flags subdocuments?


I get that by not having messages exposing them they cannot be seen from a user perspective. But wouldn't they be leaked?